### PR TITLE
NixOS & Darwin flake template testing

### PR DESCRIPTION
**Pull request by hbjydev**
This is working towards #16 -- currently contains the NixOS & Darwin components.

### CODE REVIEW

**Comment by hbjydev** [comment]
The TL;DR of this is that while, no, it doesn't _apply_ the config (i.e. by 'activating' it with `darwin-rebuild` or `nixos-rebuild`), the derivation building successfully is the _reproducible_ part of the build anyway, so by ensuring the derivation builds, we'll be able to know that there aren't any configuration errors anywhere. This will be useful for tasks like updating [home-manager](https://github.com/nix-community/home-manager), where the API can change, and because the changelogs can be hard to grok, we don't know if it'll work or not out of the box. As well as checking contributor's changes don't break the derivation outright.
